File: src/repositories/ICurrencyRepository.ts
import { ICurrency } from "../models/ICurrency";


interface ICurrencyRepository {
    create({ name, code, valueInUSD }: ICurrency): Promise<ICurrency> ;
    listAll(): Promise<ICurrency[]> ;       
    getById(_id: string): Promise<ICurrency>;
    getByCode(code: string): Promise<ICurrency> ;
    delete(_id: string): Promise<void> ;
    update({ _id, name, code, valueInUSD, updated_at }: ICurrency): Promise<void> ;
  }
  
  export { ICurrencyRepository }

File: currency-conversion/currency-conversion.Core/Interfaces/Repositories/ICurrencyRepository.cs
using currency_conversion.Core.Models;
using Microsoft.EntityFrameworkCore.ChangeTracking;

namespace currency_conversion.Core.Interfaces.Repositories
{
    public interface ICurrencyRepository
    {
        public bool Create(Currency currency);

        public void CreateMany(IEnumerable<Currency> currencies);

        public Currency? Read(string code);

        public List<Currency> ReadAll();

        public List<Currency> ReadAllNotCustom();

        public bool Update(Currency currency);

        public void UpdateMany(IEnumerable<Currency> currencies);

        public bool Delete(string code);
    }
}
